lib/rubocop/config.rb

```diff
@@ -411,9 +411,12 @@ def warn_about_unrecognized_cops(invalid_cop_names)
       end
     end
 
-    def validate_syntax_cop(valid_cop_names)
-      return unless valid_cop_names.include?('Lint/Syntax') ||
-                    valid_cop_names.include?('Syntax')
+    def validate_syntax_cop
+      syntax_config = self['Lint/Syntax'] || self['Syntax']
+      default_config = ConfigLoader.default_configuration['Lint/Syntax']
+
+      return unless syntax_config &&
+                    default_config.merge(syntax_config) != default_config
 
       raise ValidationError,
```

Comment:
Shouldn't say something more meaningful here - like "It's not possible to disable this cop" or whatever?